test/indexer/test_hca_indexer.py

```diff
@@ -1256,6 +1259,43 @@ def test_accessions_fields(self):
             self.assertEqual(['E-AAAA-00'], project['array_express_accessions'])
             self.assertEqual(['PRJNA000000'], project['insdc_study_accessions'])
 
+    def test_cell_counts(self):
+        """
+        Verify the cell counts found in project, cell_suspension, and file entities
+        """
+        # Bundles from the canned staging area, both for project 90bf705c
+        # https://github.com/HumanCellAtlas/schema-test-data/
+        bundle_fqid = self.bundle_fqid(uuid='4da04038-adab-59a9-b6c4-3a61242cc972',
+                                       version='2021-01-01T00:00:00.000000Z')
+        self._index_canned_bundle(bundle_fqid)
+        bundle_fqid = self.bundle_fqid(uuid='d7b8cbff-aee9-5a05-a4a1-d8f4e720aee7',
+                                       version='2021-01-01T00:00:00.000000Z')
+        self._index_canned_bundle(bundle_fqid)
+        hits = self._get_all_hits()
+        keys = [
+            ('projects', 'estimated_cell_count'),
+            ('cell_suspensions', 'total_estimated_cells'),
+            ('files', 'matrix_cell_count')
+        ]
+        actual_counts = NestedDict(2, int)
+        for hit in sorted(hits, key=lambda d: d['_id']):
+            entity_type, aggregate = self._parse_index_name(hit)
+            contents = hit['_source']['contents']
+            for inner_entity_type, field_name in keys:
+                for inner_entity in contents[inner_entity_type]:
+                    value = inner_entity[field_name]
+                    actual_counts[aggregate][entity_type][inner_entity_type] += value
+        entity_types = ('samples', 'bundles', 'projects', 'cell_suspensions', 'files')
+        for aggregate in True, False:
```

Comment:
Not what I had in mind. You'd have to elaborate why you deviated from my suggestion or ask if it wasn't clear.

Reply:
Using the nested keys of aggregate, entity type, and inner entity type creates a large fingerprint. In addition, since the fingerprint contains sums, almost all the leaf dictionary values are the same compared to each other. For these reasons I thought using a nested `for` loop to assert the fingerprint [leaf by leaf](https://github.com/DataBiosphere/azul/blob/2748bb9f82f321104e119b4ef1770a9b44b047f9/test/indexer/test_hca_indexer.py#L1292) would be preferred to a direct comparison of the [whole fingerprint](https://github.com/DataBiosphere/azul/blob/a2a79760eaeb659562b67fb36fd6a1b96328528a/test/indexer/test_hca_indexer.py#L1288).

Reply:
That makes sense. I think I like your approach better than what you pushed most recently. What's missing in your approach is a check for extra entries in the fingerprint. I would do that by popping from the fingerprint and asserting that it's empty.

Can you inline the `keys` variable or give it a more descriptive name?

The `entity_types` variable is initialized with a literal but I think we have a method for that somewhere.

Reply:
False comes before True. Both naturally and in our sense, contributions come before aggregates.